rotkehlchen/chain/aggregator.py

```diff
@@ -1665,7 +1692,10 @@ def add_accounts_to_all_evm(
         for account in accounts:
             existed_accounts += [(chain, account) for chain in SUPPORTED_EVM_EVMLIKE_CHAINS if account in self.accounts.get(chain)]  # noqa: E501
             # Distinguish between contracts and EOAs
-            if self.is_contract(account, SupportedBlockchain.ETHEREUM):
+            if (
+                self.is_contract(account, SupportedBlockchain.ETHEREUM) and
```

Comment:
the problem with this is that the safe may have been created only in Gnosis and say Optimism. We have to check against the given EVM chain here.